src/frameworks/fastify.js

```diff
@@ -9,22 +9,30 @@ function getValidator(validateRequest) {
     } catch (err) {
         throw new Error('Missing `uri-js` dependency. Please run "npm install uri-js" to use fastify plugin');
     }
-    let skiplist = [];
+    let skiplist = {
+        get: [],
+        post: [],
+        put: [],
+        delete: []
+    };
 
     return (pluginOptions) => {
         if (pluginOptions && pluginOptions.skiplist) {
-            skiplist = pluginOptions.skiplist.map((regexStr) => new RegExp(regexStr));
+            skiplist.get = pluginOptions.skiplist.get.map((regexStr) => new RegExp(regexStr));
+            skiplist.post = pluginOptions.skiplist.post.map((regexStr) => new RegExp(regexStr));
+            skiplist.put = pluginOptions.skiplist.put.map((regexStr) => new RegExp(regexStr));
+            skiplist.delete = pluginOptions.skiplist.delete.map((regexStr) => new RegExp(regexStr));
         }
 
         return fp(function (fastify, options, next) {
-            fastify.addHook('preValidation', validate);
+            fastify.addHook('preHandler', validate);
```

Comment:
Why is the hook changed?

Reply:
"preValidation" is the first hook where we have access to "req.body", so many fastify plugins use that hook to alter body. For example, parse "multipart/form-data" etc.
I think it makes more sense to allow them do their work and validate data only after body is parsed & changed, during next hook - "preHandler".

Reply:
@ozonep This is not related to this PR, though. I'm open to making this part more customizable, so if you want, feel free to open a separate PR, that would make hook cycle customizable, but please keep `preValidation` as default.

Reply:
Nice idea!

Reply:
Reverted this change